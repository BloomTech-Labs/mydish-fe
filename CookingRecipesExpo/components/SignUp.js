import React, {useState} from 'react';
import {
  View,
  Text,
  TextInput,
  TouchableOpacity,
  ScrollView,
  Button,
  AsyncStorage,
  Image
} from 'react-native';
import axios from 'axios'
import { KeyboardAwareScrollView } from 'react-native-keyboard-aware-scroll-view'

import styles from '../styles/signUpStyles.js'

import logo from '../assets/LogoGreen.png';


const SignUp = props => {
    const [signUp, SetSignUp] = useState({username: '', password: ''})
    const [err, setErr] = useState()

    console.log(signUp)
    
    
    const onPress = async () => {
      await axios.post('https://recipeshare-development.herokuapp.com/cooks/register', signUp)
      .then(res => console.log('response from sign up axios post', res))
      .catch(err => setErr(err))

      props.navigation.navigate('Login')
    }
    
      return (

       <KeyboardAwareScrollView>
        <View style={styles.signUp}>
          <TouchableOpacity
          onPress={() => props.navigation.navigate('Home')}
          >
            <Text style={styles.exitButton}>x</Text>
          </TouchableOpacity>
          
          <View style = {{flexDirection: 'row', justifyContent: 'center', textAlign: 'center', paddingBottom: 15}}>
            <Image source={logo} style={{width: "10%", height: "105%"}}/> 
            <Text style={styles.title}>RecipeShare</Text>
          </View>
          
          <Text style={styles.createAccountTitle}>Create Account</Text>
          <Text style={styles.emailText}>Username</Text>
          <TextInput
           style={styles.inputFeilds}
           value={signUp.username}
           onChangeText={event => SetSignUp({...signUp, username:event})}/>
           {err !=null && <Text style={{marginLeft:150, color:"red"}}>Username already exists</Text>}
           <Text style={styles.passwordText}>Password</Text>
          <TextInput
           style={styles.inputFeilds}
           value={signUp.password}
           onChangeText={event => SetSignUp({...signUp, password:event})}
           secureTextEntry={true}/>
           <View style={{flexDirection: 'row-reverse',}}>
           <TouchableOpacity
             onPress={onPress}
             style={styles.createAccountButton}
           >
             <Text style={styles.createAccountText}>Create Account</Text>
           </TouchableOpacity>
           </View>
        </View>
       </KeyboardAwareScrollView>
      );
    }
    

export default SignUp;