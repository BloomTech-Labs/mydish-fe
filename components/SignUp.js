import React, { useState, useRef, useEffect } from 'react';
import {
  View,
  Text,
  TextInput,
  TouchableOpacity,
  SafeAreaView,
  Alert,
  KeyboardAvoidingView,
  ScrollView,
  ImageBackground,
} from 'react-native';
import { useDispatch, useSelector } from 'react-redux';
import { clearError, registerUser } from '../store/auth/authActions';
import styles from '../styles/authPageStyles';
import backgroundImg from '../assets/auth-page-background.jpg';
import { maxUsername } from '../constants/maxLength';
import AuthButton from './AuthButton';
import { logoHeaderPlain } from './header/navigationHeader';

const SignUp = ({ navigation }) => {
  const [credentials, setSignUp] = useState({
    username: '',
    password: '',
    confirmPassword: '',
    email: '',
  });
  const errorMsg = useSelector((state) => state.auth.error);
  const dispatch = useDispatch();
  const usernameInput = useRef(null);
  const passwordInput = useRef(null);
  const confirmPasswordInput = useRef(null);
  const emailInput = useRef(null);

  useEffect(() => {
    if (errorMsg != null) {
      emptyFieldsAlert();
    }
  }, [errorMsg]);

  function validateEmail(email) {
    var re = /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
    return re.test(String(email).toLowerCase());
  }

  const register = async () => {
    if (credentials.password != credentials.confirmPassword) {
      return Alert.alert("Passwords don't match");
    }
    if (!validateEmail(credentials.email)) {
      return Alert.alert('Please enter a valid email address');
    }
    const success = await dispatch(registerUser(credentials));
    console.log(credentials);
    if (success) {
      navigation.navigate('App');
    }
  };

  const emptyFieldsAlert = () => {
    return Alert.alert('Oops!', 'Please provide a username and password.', [
      { title: 'Okay' },
    ]);
  };

  return (
    <SafeAreaView>
      <ImageBackground source={backgroundImg} style={styles.backgroundImg}>
        <KeyboardAvoidingView behavior="height">
          <ScrollView>
            <View style={styles.container}>
              <View style={styles.contentContainer}>
                <Text style={styles.inputLabelText}>Username</Text>
                <TextInput
                  ref={usernameInput}
                  style={styles.inputFields}
                  maxLength={maxUsername}
                  value={credentials.username}
                  returnKeyType="next"
                  onSubmitEditing={() => passwordInput.current.focus()}
                  onChangeText={(event) =>
                    setSignUp({
                      ...credentials,
                      username: event,
                    })
                  }
                />
                <Text
                  style={styles.maxLengthIndicator}
                >{`${credentials.username.length}/${maxUsername}`}</Text>

                <Text style={styles.inputLabelText}>Password</Text>
                <TextInput
                  ref={passwordInput}
                  style={styles.inputFields}
                  value={credentials.password}
                  returnKeyType="next"
                  onChangeText={(event) =>
                    setSignUp({
                      ...credentials,
                      password: event,
                    })
                  }
                  secureTextEntry={true}
                  onSubmitEditing={() => confirmPasswordInput.current.focus()}
                />

                <Text style={styles.inputLabelText}>Confirm Password</Text>
                <TextInput
                  ref={confirmPasswordInput}
                  style={styles.inputFields}
                  value={credentials.confirmPassword}
                  returnKeyType="next"
                  onChangeText={(event) =>
                    setSignUp({
                      ...credentials,
                      confirmPassword: event,
                    })
                  }
                  secureTextEntry={true}
                  onSubmitEditing={() => emailInput.current.focus()}
                />

                <Text style={styles.inputLabelText}>Email</Text>
                <TextInput
                  ref={emailInput}
                  style={styles.inputFields}
                  value={credentials.email}
                  returnKeyType="go"
                  onChangeText={(event) =>
                    setSignUp({
                      ...credentials,
                      email: event,
                    })
                  }
                  onSubmitEditing={register}
                />

                <View style={styles.promptContainer}>
                  <Text style={styles.questionPrompt}>Have an account? </Text>
                  <TouchableOpacity
                    onPress={() => {
                      dispatch(clearError());
                      navigation.navigate('Login');
                    }}
                  >
                    <Text style={styles.switchAuthPageLink}>Sign in</Text>
                  </TouchableOpacity>
                </View>

                <View style={{ flexDirection: 'row-reverse' }}>
                  <AuthButton
                    parent="signup"
                    credentials={credentials}
                    register={register}
                  />
                </View>
              </View>
            </View>
          </ScrollView>
        </KeyboardAvoidingView>
      </ImageBackground>
    </SafeAreaView>
  );
};
SignUp.navigationOptions = logoHeaderPlain;

export default SignUp;
