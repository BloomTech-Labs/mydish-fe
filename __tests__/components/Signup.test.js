import React from 'react';
import renderer from 'react-test-renderer';
import SignUp from '../../components/SignUp';
import { createStore } from 'redux';
import { Provider } from 'react-redux';

test('matches snapshot', () => {
  // The SignUp component only needs the error portion of the store,
  //     so we create a simpler store doing just that:
  const store = createStore(() => ({ auth: { error: null } }));

  // Then we create our smaller SignUp component with the provider
  const renderTree = (
    <Provider store={store}>
      <SignUp />
    </Provider>
  );
  const signupSnap = renderer.create(renderTree).toJSON();
  expect(signupSnap).toMatchSnapshot();
});
