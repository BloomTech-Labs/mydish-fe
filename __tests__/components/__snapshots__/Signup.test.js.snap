// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
>
  <Image
    source={1}
    style={
      Object {
        "maxHeight": "90%",
        "maxWidth": "80%",
        "position": "absolute",
        "resizeMode": "cover",
        "zIndex": -10000,
      }
    }
  />
  <View
    onLayout={[Function]}
  >
    <RCTScrollView>
      <View>
        <View
          style={
            Object {
              "height": "100%",
              "paddingHorizontal": 50,
              "paddingTop": 70,
              "width": "100%",
            }
          }
        >
          <View
            style={
              Object {
                "display": "flex",
                "flexDirection": "column",
                "height": "65%",
                "justifyContent": "space-between",
                "paddingVertical": 70,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#FFFFFF",
                  "marginBottom": 5,
                }
              }
            >
              Username
            </Text>
            <TextInput
              allowFontScaling={true}
              maxLength={15}
              onChangeText={[Function]}
              onSubmitEditing={[Function]}
              rejectResponderTermination={true}
              returnKeyType="next"
              style={
                Object {
                  "backgroundColor": "#FFFFFF",
                  "borderColor": "#D4D4D4",
                  "borderRadius": 6,
                  "borderWidth": 1,
                  "color": "#959595",
                  "height": 40,
                  "marginBottom": 25,
                  "minHeight": "5%",
                  "paddingLeft": 10,
                }
              }
              underlineColorAndroid="transparent"
              value=""
            />
            <Text
              style={
                Object {
                  "alignSelf": "flex-end",
                  "color": "#FFFFFF",
                  "fontSize": 12,
                  "marginTop": -20,
                }
              }
            >
              0/15
            </Text>
            <Text
              style={
                Object {
                  "color": "#FFFFFF",
                  "marginBottom": 5,
                }
              }
            >
              Password
            </Text>
            <TextInput
              allowFontScaling={true}
              onChangeText={[Function]}
              onSubmitEditing={[Function]}
              rejectResponderTermination={true}
              returnKeyType="go"
              secureTextEntry={true}
              style={
                Object {
                  "backgroundColor": "#FFFFFF",
                  "borderColor": "#D4D4D4",
                  "borderRadius": 6,
                  "borderWidth": 1,
                  "color": "#959595",
                  "height": 40,
                  "marginBottom": 25,
                  "minHeight": "5%",
                  "paddingLeft": 10,
                }
              }
              underlineColorAndroid="transparent"
              value=""
            />
            <View
              style={
                Object {
                  "flexDirection": "row",
                  "marginTop": -20,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#FFFFFF",
                  }
                }
              >
                Have an account?
                 
              </Text>
              <View
                accessible={true}
                isTVSelectable={true}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "#D2291F",
                      "fontSize": 14,
                      "fontWeight": "bold",
                      "textDecorationLine": "underline",
                    }
                  }
                >
                  Login
                </Text>
              </View>
            </View>
            <View
              style={
                Object {
                  "flexDirection": "row-reverse",
                }
              }
            >
              <View
                accessible={true}
                isTVSelectable={true}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignSelf": "center",
                    "backgroundColor": "#D2291F",
                    "borderRadius": 5,
                    "height": 40,
                    "justifyContent": "center",
                    "marginTop": 30,
                    "opacity": 1,
                    "paddingVertical": 10,
                    "width": 150,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "#FFFFFF",
                      "fontSize": 14,
                      "fontWeight": "bold",
                      "textAlign": "center",
                    }
                  }
                >
                  Sign Up
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</RCTSafeAreaView>
`;
